### DISCUSSION TITLE
Course Titles

### DISCUSSION THREAD

**Discussion by hodahakim** [NRTC/quick-start-guide#83] [Team Posts]
______________________________
## 1. History of Computer   
_______________________________
## 2. Computer Hardware:                                :point_right:  @S-hajipour   @firouzian 

- ISA (Instruction Set Architecture)
- comparison of in instruction set of architection:
	- X86
	- RISC V
- Floating Point (IEEE-754)
- Parallel Computing (Distribute Memory vs. Share Memory)
- Multi-Core
- PCI Express (Peripheral Component Interconnect Express)
- BIOS (Startup Firmware):
	- UEFI
	- Legacy
________________________________
## 3. OS (Operating System):                                           :point_right:  @hodahakim 

- Multi-tasking
- kernel
	- LINUX (UNIX-Like)
	- BSD
	- NT
	- UNIX
- xorg
- libC standard library
- compiler
- GNU (GNU Operating System)
- Portable Operating System Interface (POSIX)
- Filesystem Hierarchy Standard (FHS)
- GNU GRUB (GNU GRand Unified Bootloader)
- GNU coreutils

____________________________________
## 4. GNU/Linux Distribution:                                           :point_right: @elmiradehghan 

- Debian
- Gentoo
- Fedora

____________________________________
## 5. Compilers:                                                           :point_right: @elmiradehghan 

- FORTRAN Compilers
- gcc (GNU C Compiler)
- GCC (GNU compile collection)
- Compiled Language (C, FORTRAN, ...) VS. Interpreted Languages (Python, Ruby, ...)

____________________________________
## 6. IPC (Inter Process Communication):                             :point_right:  @S-hajipour 
- POSIX Signal:
	- SIGINT (signal interrupt), ctrl+c
	- SIGHUB
	- SIGKILL
	- SIGSTOP

____________________________________
## 7. Standard file format for executables:           :point_right:  @hodahakim 

- ELF (The Executable and Linkable Format) for unix-like systems
- PE  (The Portable Executable format ) for windows operating systems

____________________________________
## 8. Uinx-like File System:

- look like tree / (The root of the file system tree)
- File System
	- EXT4
	- BTRFS
	- ZFS

____________________________________
## 9. User Interfaces (UI):                                                 :point_right:    @elmiradehghan 

- GUI (graphical user interface):
	- Desktop Enviroments:
		- KDE
		- GNOME
- TUI (text-based user interface)
- CLI (command-line interface)

____________________________________
## 10. Version Control System (VCS):                              :point_right:        @hodahakim 

- git:
	- .git (Hidden File)
	- stage
	- commit
	- repository
	- push
	- pull
	- merge
	- fork
	- branch
- mercurial
- cvs

____________________________________
## 11. How to use:                                                            :point_right:    @elmiradehghan 

- manpages (manual page):
	- man command
	- online manpage
- regex (Regular Expression)
- patterns
- stdout/stdin/stderr
- redirectors (pipe, >, <)
- grep
- Script File:
	- batch file  (in Microsoft Windows)
		PowerShell
		cmd (Command Prompt)
	- shell script (in Unix-like operating systems)
		shell
		terminal
- dpkg/apt
- Environment variables

____________________________________
## 12. Build from Source:                                                 :point_right:  @hodahakim 

- build automation:
	- GNU make
	- Cmake

____________________________________
## 13. Miscellaneous:                                          :point_right: @elmiradehghan 

- licenses
- free software
- Application binary interface (ABI)
- Application Programming interface (API)
- Difference between API and ABI
- WineHQ project
- system monitor
- containerization/virtualization
- apache web server
- SQL
- Redundancy
____________________________________
## 14. Numerical Analysis:                                     :point_right:        @elmiradehghan 

- Butterfly effect; Numerical Stability
- Difference between numeric and symbolic computation

___________________________________
## 15. Vagrant                                                                   :point_right:  @hodahakim 
____________________________________

**Comment by hodahakim** [comment]
@S-hajipour  @elmiradehghan 

<div align="right">

سلام
 این سرفصل ها را از   

[Trello](https://trello.com/c/nvYAU02W/13-course-titles)  
به اینجا منتقل کردم
فعلا بر اساس  پول ریکوئست های داده شده اسم نویسنده هر بخش را روبروی عنوان آن منشن کردم
این متن توسط همه ما قابل ویرایش است
بنابراین هر کس می تواند به همین ترتیب، سرفصلی را که در حال نوشتن آن است، با منشن کردن اسمش درمقابل آن، مشخص کند
سرفصل های جدیدی را هم که به نظرتان می رسد به عناوین دوره اضافه نشده با همین فرمتِ 
[Markdown](https://www.markdownguide.org/basic-syntax/) 
اضافه  کنید

</div>

**Comment by S-hajipour** [comment]
سلام 
6. IPC (Inter Process Communication):
رو به من ارجاع بدید
